internal/telemetry/log/log.go

```diff
@@ -46,7 +44,7 @@ func Error(format string, args ...any) {
 
 func log(lvl telemetry.LogLevel, format string, args []any) {
 	opts, fmtArgs := divideArgs(args)
-	telemetry.Log(lvl, fmt.Sprintf(format, fmtArgs...), opts...)
+	telemetry.Log(lvl, format, opts...)
```

Comment:
This wouldn't solve the issue if developer decides to send a string that is already formatted.

I'm working on a static check to prevent abusing the `telemetry.Log`.

Ideally, we need a scrubbing/sanitizing logger as a last step and with some heuristic (best-effort) drops the logs that have some possible sensitive data.

Reply:
The mentioned static checks are part of https://github.com/DataDog/dd-trace-go/pull/3706